src/NHibernate/Proxy/AbstractLazyInitializer.cs

```diff
@@ -273,11 +274,9 @@ private object GetProxyOrNull()
 			return null;
 		}
 		
-		private void SetReadOnly(bool readOnly)
+		private void SetReadOnly(IPersistenceContext persistenceContext, bool readOnly, IEntityPersister entityPersister)
```

Comment:
Ideally, we should not have to add arguments to a method only for sparing some property calls. That is why I was thinking about removing the check in the property, for turning it back to an innocuous call. (As should be any property call.)

Now it seems a lot more safer to do it this way, as removing the check need a lot of analysis for ensuring the check will in all cases have already be done some point sooner. (And it could imply adding that check to many places where it was previously done through the property call.)